working/0723-static-extensions/feature-specification-variant2.md

```diff
@@ -0,0 +1,347 @@
+name clash prevents an implicitly resolved invocation. However, implicitly
+resolved invocations are specified in the rest of this section by reducing
+them to explicitly resolved ones.*
+
+A constructor invocation of the form `C<T1 .. Tm>.name(args)` is partially
+resolved by looking up a constructor named `C.name` in the class `C` and in
+every accessible extension with on-class `C`. A compile-time error
+occurs if no such constructor is found. Similarly, an invocation of the
+form `C<T1 ... Tm>(args)` uses a lookup for constructors named `C`.
+
+If a constructor in `C` with the requested name was found, the pre-feature
+static analysis and dynamic semantics apply. *That is, the class always
+wins.*
+
+Otherwise, the invocation is partially resolved to a set of candidate
+constructors found in extensions. Each of the candidates _kj_ is
+vetted as follows:
+
+Assume that _kj_ is a constructor declared by a extension _D_ named
+`E` with type parameters `X1 extends B1 .. Xs extends Bs` and on-type
+`C<S1 .. Sm>`.  Find actual values `U1 .. Us` for `X1 .. Xs` satisfying the
+bounds `B1 .. Bs`, such that `([U1/X1 .. Us/Xs]C<S1 .. Sm>) == C<T1 .. Tm>`.
+If this fails then remove _kj_ from the set of candidate constructors.
+Otherwise note that _kj_ uses actual type arguments `U1 .. Us`.
+
+If all candidate constructors have been removed, or more than one candidate
+remains, a compile-time error occurs. Otherwise, the invocation is
+henceforth treated as `E<U1 .. Us>.C<T1 .. Tm>.name(args)` (respectively
+`E<U1 .. Us>.C<T1 .. Tm>(args)`).
+
+A constructor invocation of the form `C.name(args)` (respectively
+`C(args)`) where `C` resolves to a non-generic class is resolved in the
+same manner, with `m == 0`. *In this case, type parameters declared by `E`
+will be bound to values selected by instantiation to bound.*
+
+Consider a constructor invocation of the form `C.name(args)` (and similarly
+for `C(args)`) where `C` resolves to a generic class. As usual, the
+invocation is treated as in the pre-feature language when it resolves to a
+constructor declared by the class `C`.
+
+In the case where the context type schema for this invocation fully
+determines the actual type arguments of `C`, the expression is changed to
+receive said actual type arguments, `C<T1 .. Tm>.name(args)`, and treated
+as described above.
+
+In the case where the invocation resolves to exactly one constructor
+`C.name` (or `C`) declared by a extension named `E`, the invocation
+is treated as `E.C.name(args)` (respectively `E.C(args)`).
+
+Otherwise, when there are two or more candidates from extensions,
+an error occurs. *We do not wish to specify an approach whereby `args` is
+subject to type inference multiple times, and hence we do not support type
+inference for `C.name(args)` in the case where there are multiple distinct
+declarations whose signature could be used during the static analysis of
+that expression.*
+
+### Dynamic Semantics
```

Comment:
Runtime semantics.

The word "dynamic" is loaded in Dart, don't use it for anything but `dynamic`.

Reply:
The phrase 'dynamic semantics' is used to denote the dynamic or run-time semantics in the language specification. There are no occurrences of 'run-time semantics' in normative text, and only one occurrence in commentary.

We could change this, but it makes no sense to say that we should use non-standard terminology in just one feature specification, that should be a global operation.